src/main/scala/io/github/paoloboni/binance/common/parameters/Depth.scala

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2021 Paolo Boni
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package io.github.paoloboni.binance.common.parameters
+
+final case class Depth(symbol: String, limit: Int)
```

Comment:
Based on the API spec, the limit could be modelled as enum for both version 2.x and 3.x of Scala.
Specs:

Limit | Weight
-- | --
5, 10, 20, 50, 100 | 1
500 | 5
1000 | 10
5000 | 50

Scala 3 (under scala-3 directory):
```
enum DepthLimit(val weight: Int) derives EnumDecoder:
  case 5 extends DepthLimit(1)
  ...
  case 500 extends DepthLimit(5)
  ...
```
Scala 2 (under scala-2 directory):
```
sealed abstract class DepthLimit(val weight: Int) extends EnumEntry
object Interval extends Enum[DepthLimit] with CirceEnum[DepthLimit] {
  val values: immutable.IndexedSeq[DepthLimit] = findValues

  case object `5` extends DepthLimit(1)
  ...
  case object `500` extends DepthLimit(5)
  ...
}
```

Reply:
@paoloboni Done.